wordpress-mobile/WordPress-Android
# Issue/8640 add migration dialog

### Fix
Add a migration dialog shown on the ***Sites*** tab when Quick Start is in progress as described in https://github.com/wordpress-mobile/WordPress-Android/issues/8640.  The migration dialog is shown only once for all sites.  See the screenshots below for illustration.

![v2_sites_migration](https://user-images.githubusercontent.com/3827611/49099793-f3ab1a00-f237-11e8-8e70-d804139e52c6.png)

#### Note
There are two `// TODO: Quick Start - ` items included in these changes that are related to analytics.  They will be addressed in a separate pull request as described in https://github.com/wordpress-mobile/WordPress-Android/issues/8649.

### Test
1. Go to ***Sites*** tab.
2. Tap ***Switch Site*** button in top card view.
3. Tap site with Quick Start in progress.
4. Notice migration dialog is shown.
5. Rotate device.
6. Notice migration dialog is shown.
7. Dismiss migration dialog.
8. Tap ***Switch Site*** button in top card view.
9. Tap site without Quick Start in progress.
10. Notice migration dialog is not shown.
11. Tap ***Switch Site*** button in top card view.
12. Tap site with Quick Start in progress.
13. Notice migration dialog is not shown.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.
